packages/dom-helper/lib/main.js

```diff
@@ -565,15 +567,39 @@ prototype.parseHTML = function(html, contextualElement) {
 };
 
 var parsingNode;
+var URL;
 
-// Used to determine whether a URL needs to be sanitized.
 prototype.protocolForURL = function(url) {
+  // Test to see if our DOM implementation parses
+  // and normalizes URLs.
+  var protocol = browserProtocolForURL.call(this, 'foobar:baz');
+  if (protocol === 'foobar:') {
+    // Swap in the method that doesn't do this test now that
+    // we know it works.
+    prototype.protocolForURL = browserProtocolForURL;
+  } else {
+    // Otherwise, we need to fall back to our own URL parsing.
+    // Global `require` is shadowed by Ember's loader so we have to use the fully
+    // qualified `module.require`.
+    URL = module.require('url');
+    prototype.protocolForURL = nodeProtocolForURL;
+  }
+
+  return this.protocolForURL(url);
+};
+
+function browserProtocolForURL(url) {
   if (!parsingNode) {
     parsingNode = this.document.createElement('a');
```

Comment:
then it can be done in the constructor, and rather then overriding the method on the prototype, it could defer to a delegate which changes. This prevents some sadness we have been seeing with jits, and swapping out methods at runtime.


```js

function C() {
   this._prococalForURL = ...;
}

C.prototype.prococalForURL = function(url) {
  this._prococalForURL(url)
}
```